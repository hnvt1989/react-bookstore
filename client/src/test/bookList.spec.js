import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import { shallow } from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import BookList from "../components/books/BookList";

const mockStore = configureMockStore();
const store = mockStore({});

//disableLifecycleMethods is auto set to True in Enzyme 3
Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: true });

//use jest -u to update the snapshot
//using react-render + jest snapshot feature
test("Books Listing header", () => {
  const bookListComponent = shallow(
    <Provider store={store}>
      <BookList />
    </Provider>
  );

  expect(bookListComponent.children('h2').text()).toEqual('Books');

  //const header = bookListComponent.children('div h2');
  //const header = bookListComponent.find("div h2");
  // const header = bookListComponent.find('h2');
  // expect(header.text()).toEqual('Books');

});