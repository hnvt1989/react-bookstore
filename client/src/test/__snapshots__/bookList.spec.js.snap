// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Create component 1`] = `
<div>
  <h3>
    Create a Book
  </h3>
  <form
    className="ui form error"
    onSubmit={[Function]}
  >
    <div
      className="field "
    >
      <label>
        Enter Name
      </label>
      <input
        autoComplete="off"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        value=""
      />
    </div>
    <div
      className="field "
    >
      <label>
        Enter Description
      </label>
      <input
        autoComplete="off"
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        value=""
      />
    </div>
    <div>
      <div
        className="ui button"
        onClick={[Function]}
      >
        Add Item
      </div>
      <br />
    </div>
    <div>
      <button
        className="ui button primary"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`Books Listing 1`] = `
<div>
  <h3>
    Create a Book
  </h3>
  <form
    className="ui form error"
    onSubmit={[Function]}
  >
    <div
      className="field "
    >
      <label>
        Enter Name
      </label>
      <input
        autoComplete="off"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        value=""
      />
    </div>
    <div
      className="field "
    >
      <label>
        Enter Description
      </label>
      <input
        autoComplete="off"
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        value=""
      />
    </div>
    <div>
      <div
        className="ui button"
        onClick={[Function]}
      >
        Add Item
      </div>
      <br />
    </div>
    <div>
      <button
        className="ui button primary"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
